src/main/drivers/system_stm32g4xx.c

```diff
@@ -0,0 +1,207 @@
+/*
+ * This file is part of Cleanflight.
+ *
+ * Cleanflight is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Cleanflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Cleanflight.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <string.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+#include "platform.h"
+
+#include "drivers/accgyro/accgyro_mpu.h"
+#include "drivers/exti.h"
+#include "drivers/nvic.h"
+#include "drivers/memprot.h"
+#include "drivers/persistent.h"
+#include "drivers/system.h"
+
+bool isMPUSoftReset(void)
+{
+    if (cachedRccCsrValue & RCC_CSR_SFTRSTF)
+        return true;
+    else
+        return false;
+}
+
+void systemInit(void)
+{
+    memProtReset();
+    memProtConfigure(mpuRegions, mpuRegionCount);
+
+    // Configure NVIC preempt/priority groups
+    HAL_NVIC_SetPriorityGrouping(NVIC_PRIORITY_GROUPING);
+
+    // cache RCC->RSR value to use it in isMPUSoftReset() and others
+    cachedRccCsrValue = RCC->CSR;
+
+    /* Accounts for OP Bootloader, set the Vector Table base address as specified in .ld file */
+    //extern void *isr_vector_table_base;
+    //NVIC_SetVectorTable((uint32_t)&isr_vector_table_base, 0x0);
+    //__HAL_RCC_USB_OTG_FS_CLK_DISABLE;
+
+    //RCC_ClearFlag();
+
+    // Init cycle counter
+    cycleCounterInit();
+
+    // SysTick is updated whenever HAL_RCC_ClockConfig is called.
+
+    // Disable UCPD pull-down functionality on PB4
+    // AN5093 6.3.3:
+    // (after device startup this pull-down on PB4 can be disabled by
+    // setting UCPD1_DBDIS bit in the PWR_CR3 register).
+    //
+    // There is also a similar UCPD stand-by functionality control,
+    // but don't fiddle with this one. In particular, disabling it by
+    // calling HAL_PWREx_DisableUSBStandByModePD() will cause boot loader
+    // invocation to fail.
+
+#if defined(PWR_CR3_UCPD_DBDIS)
+    HAL_PWREx_DisableUSBDeadBatteryPD();
+#endif
+}
+
+void systemReset(void)
+{
+    // SCB_DisableDCache();
+    // SCB_DisableICache();
+
+    __disable_irq();
+    NVIC_SystemReset();
+}
+
+void forcedSystemResetWithoutDisablingCaches(void)
+{
+    persistentObjectWrite(PERSISTENT_OBJECT_RESET_REASON, RESET_FORCED);
+
+    __disable_irq();
+    NVIC_SystemReset();
+}
+
+void systemResetToBootloader(bootloaderRequestType_e requestType)
+{
+    switch (requestType) {
+#if defined(USE_FLASH_BOOT_LOADER)
```

Comment:
Not sure this is needed here - are there plans to release boards with this MCU designed to store the firmware in external flash?

Reply:
Nope. It should be removed.